src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -128,7 +128,15 @@ private static MethodSymbol GetFlushPayload(CSharpCompilation compilation, CShar
         private static bool IsTestMethod(MethodSymbol method)
         {
             // PROTOTYPE (https://github.com/dotnet/roslyn/issues/9811): Make this real. 
-            return method.Name.StartsWith("Test");
+            var attributes = method.GetAttributes();
+            foreach (var attribute in attributes)
+            {
+                if (attribute.IsTargetAttribute("Xunit", "FactAttribute"))
```

Comment:
As noted in the conversation this is a prototype line.  We should reference the actual bug tracking the work here:

https://github.com/dotnet/roslyn/issues/9811